sdk/core/azure-core/src/main/java/com/azure/core/http/policy/RetryPolicy.java

```diff
@@ -38,7 +38,19 @@ public RetryPolicy() {
      * @param retryStrategy The {@link RetryStrategy} used for retries.
      */
     public RetryPolicy(RetryStrategy retryStrategy) {
-        this.retryStrategy = Objects.requireNonNull(retryStrategy, "'retryStrategy' cannot be null");
+        Objects.requireNonNull(retryStrategy, "'retryStrategy' cannot be null");
+        this.retryPolicyOptions = new RetryPolicyOptions().setRetryStrategy(retryStrategy);
+    }
+
+    /**
+     * Creates a {@link RetryPolicy} with the provided {@link RetryPolicyOptions}.
+     *
+     * @param retryPolicyOptions with given {@link RetryPolicyOptions}.
+     * @throws NullPointerException if {@code retryPolicyOptions} is {@code null}.
+     */
+    public RetryPolicy(RetryPolicyOptions retryPolicyOptions) {
+        this.retryPolicyOptions = Objects.requireNonNull(retryPolicyOptions,
```

Comment:
This should also check that `retryPolicyOptions.getRetryStrategy()` is also non-null. We need that to be set.